File: FlowCrypt/src/main/java/com/flowcrypt/email/ui/widget/PGPContactChipSpan.java
/*
 * © 2016-2018 FlowCrypt Limited. Limitations apply. Contact human@flowcrypt.com
 * Contributors: DenBond7
 */

package com.flowcrypt.email.ui.widget;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.hootsuite.nachos.chip.ChipSpan;

/**
 * This class describes the representation of {@link ChipSpan} with PGP existing.
 *
 * @author Denis Bondarenko
 * Date: 15.08.2017
 * Time: 16:28
 * E-mail: DenBond7@gmail.com
 */

public class PGPContactChipSpan extends ChipSpan {
  private Boolean isHasPgp;

  public PGPContactChipSpan(@NonNull Context context, @NonNull CharSequence text, @Nullable
      Drawable icon, Object
                                data) {
    super(context, text, icon, data);
  }

  public PGPContactChipSpan(@NonNull Context context, @NonNull PGPContactChipSpan
      pgpContactChipSpan) {
    super(context, pgpContactChipSpan);
    this.isHasPgp = pgpContactChipSpan.isHasPgp();
  }

  public Boolean isHasPgp() {
    return isHasPgp;
  }

  public void setHasPgp(Boolean hasPgp) {
    isHasPgp = hasPgp;
  }
}

File: FlowCrypt/src/main/java/com/flowcrypt/email/ui/widget/PGPContactChipSpan.kt
/*
 * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
 * Contributors: DenBond7
 */

package com.flowcrypt.email.ui.widget

import android.content.Context
import android.content.res.ColorStateList
import android.graphics.drawable.Drawable

import com.hootsuite.nachos.chip.ChipSpan

/**
 * This class describes the representation of [ChipSpan] with PGP existing.
 *
 * @author Denis Bondarenko
 * Date: 15.08.2017
 * Time: 16:28
 * E-mail: DenBond7@gmail.com
 */
class PGPContactChipSpan : ChipSpan {
  var hasAtLeastOnePubKey: Boolean? = false
  var hasNotExpiredPubKey: Boolean? = false
  var hasUsablePubKey: Boolean? = false
  var hasNotRevokedPubKey: Boolean? = false

  /**
   * The last modified value that saved after [setBackgroundColor]. Can be null
   */
  var chipBackgroundColor: ColorStateList? = null

  constructor(context: Context, text: CharSequence, icon: Drawable?, data: Any?) : super(
    context,
    text,
    icon,
    data
  )

  constructor(context: Context, pgpContactChipSpan: PGPContactChipSpan) : super(
    context,
    pgpContactChipSpan
  ) {
    this.hasAtLeastOnePubKey = pgpContactChipSpan.hasAtLeastOnePubKey
    this.hasNotExpiredPubKey = pgpContactChipSpan.hasNotExpiredPubKey
    this.hasUsablePubKey = pgpContactChipSpan.hasUsablePubKey
    this.hasNotRevokedPubKey = pgpContactChipSpan.hasNotRevokedPubKey
  }

  override fun setBackgroundColor(backgroundColor: ColorStateList?) {
    super.setBackgroundColor(backgroundColor)
    chipBackgroundColor = backgroundColor
  }
}
